pkg/subctl/cmd/resource.go

```diff
@@ -95,3 +97,13 @@ func getGatewaysResource(config *rest.Config) *submarinerclientv1.GatewayList {
 
 	return gateways
 }
+
+func getBrokerRestConfig(submariner *v1alpha1.Submariner) (*rest.Config, error) {
+	tlsClientConfig := rest.TLSClientConfig{}
+
```

Comment:
@skitt Similar code in admiral does this:

```
          if useTokenCAForEndpoint {
                caDecoded, err := base64.StdEncoding.DecodeString(brokerSpec.Ca)
                if err != nil {
                        return nil, "", fmt.Errorf("error decoding CA data: %v", err)
                }
                tlsClientConfig.CAData = caDecoded
        }
```

and the caller first tries w/o the token then with it if the first try fails. I'm not really clear on this. Do we need to do the same or is this OK?

Reply:
Probably we need to do the same.

Some endpoints work just with the CA, some with both, I believe this was due to a difference in rancher endpoints. But I can't recall the exact details now.

We could not do it, but in that case, can you leave a comment/link here so we have a reference if problems arise we can investigate & document the issue better.

Reply:
Will do. Waiting on @skitt to weight in. If we need to do exactly what admiral is doing it would make sense to reuse that but it would need an admiral patch we well.

Reply:
Yes, I can’t remember the details right now but IIRC we had to do this for OpenShift on AWS as well in some setups.

Reply:
So should we just always use the CA?

Reply:
We need the check, see https://github.com/submariner-io/submariner/commit/dc68b2c3c97ab5cfcf89c7a2364495fd5806dffb and https://github.com/submariner-io/submariner/issues/649 for details.

Reply:
OK - https://github.com/submariner-io/admiral/pull/199